containers/podman-desktop-extension-ai-lab
# feat: start REST server

### What does this PR do?

1. Starts an API server on port 10434 (or a random one if this port is in use).

Serves two endpoints:

- `GET /spec`: returns the spec in OpenAPI YAML format 
- `GET /api/version`: returns the version of AI Lab

### Validation

Any endpoints (below /api for now - related to discussions points 1 & 2) are validated:
- `curl http://localhost:10434/api/not_found` should return an error
- `curl http://localhost:10434/api/version?bad_param` should return an error  
- modify the response in `getVersion` to add a new property to the returned object, or to rename the `version` property, `curl http://localhost:10434/api/version` should then display logs (at error level) that the response is incorrect 

### To discuss

- Do we want to serve the spec at `/spec` or at `/api/spec`?
- Do we want this endpoint to be documented in the OpenAPI Spec ?

### Screenshot / video of UI



### What issues does this PR fix or reference?

Fixes #1295



### How to test this PR?

- install the extension
  - with image `quay.io/jeffmaury/ai-lab:rest-server`
  - or, by building the image:
    ```
    $ yarn
    $ yarn build
    $ podman build . -t your_image
    $ podman push your_image
    ```
- start Podman Desktop with the AI Lab extension
- check in the status bar that a message appears indicating that AI Lab is listening on port
- click the status
- copy the URL from the message box
- access the different endpoints from this URL
